drivers/clock_control/clock_control_nrf2_auxpll.c

```diff
@@ -0,0 +1,254 @@
+		return rc;
+	}
+
+	const struct clock_control_nrf_auxpll_config *config = dev->config;
+
+	nrf_auxpll_ctrl_frequency_set(config->auxpll, config->frequency);
+
+	nrf_auxpll_lock(config->auxpll);
+	nrf_auxpll_trim_ctune_set(config->auxpll, sys_read8(config->ficr_ctune));
+	nrf_auxpll_config_set(config->auxpll, &config->cfg);
+	nrf_auxpll_ctrl_outsel_set(config->auxpll, config->out_div);
+	nrf_auxpll_unlock(config->auxpll);
+
+	nrf_auxpll_ctrl_mode_set(config->auxpll, NRF_AUXPLL_CTRL_MODE_LOCKED);
+
+	return 0;
+}
+
+static DEVICE_API(nrf_clock_control, drv_api_auxpll) = {
+	.std_api = {
+		.on = api_nosys_on_off,
+		.off = api_nosys_on_off,
+		.get_rate = clock_control_nrf_auxpll_get_rate,
+		.get_status = clock_control_nrf_auxpll_get_status,
+	},
+	.request = api_request_auxpll,
+	.release = api_release_auxpll,
+	.cancel_or_release = api_cancel_or_release_auxpll,
+};
+
+#define CLOCK_CONTROL_NRF_AUXPLL_DEFINE(n)													  \
+	BUILD_ASSERT(																			  \
+		DT_INST_PROP(n, nordic_frequency) == NRF_AUXPLL_FREQUENCY_DIV_MIN	 ||          	  \
+		DT_INST_PROP(n, nordic_frequency) == NRF_AUXPLL_FREQUENCY_AUDIO_44K1 ||          	  \
+		DT_INST_PROP(n, nordic_frequency) == NRF_AUXPLL_FREQUENCY_USB_24M	 ||          	  \
+		DT_INST_PROP(n, nordic_frequency) == NRF_AUXPLL_FREQUENCY_AUDIO_48K  ||          	  \
+		DT_INST_PROP(n, nordic_frequency) == NRF_AUXPLL_FREQUENCY_DIV_MAX,              	  \
+		"Invalid nordic,frequency value in DeviceTree for AUXPLL instance " #n);      		  \
+    static struct dev_data_auxpll data_auxpll##n    = { 									  \
+	.dev = DEVICE_DT_INST_GET(n), 														  	  \
+	}; 																					      \
+	static const struct clock_control_nrf_auxpll_config config##n = {                         \
+		.auxpll = (NRF_AUXPLL_Type *)DT_INST_REG_ADDR(n),                                     \
+		.ref_clk_hz = DT_PROP(DT_INST_CLOCKS_CTLR(n), clock_frequency),                       \
+		.ficr_ctune = DT_REG_ADDR(DT_INST_PHANDLE(n, nordic_ficrs)) +                         \
+			      DT_INST_PHA(n, nordic_ficrs, offset),                                       \
+		.cfg =                                                                                \
+			{                                                                                 \
+				.outdrive = DT_INST_PROP(n, nordic_out_drive),                                \
+				.current_tune = DT_INST_PROP(n, nordic_current_tune),	                      \
+				.sdm_off = DT_INST_PROP(n, nordic_sdm_disable),                               \
+				.dither_off = DT_INST_PROP(n, nordic_dither_disable),                         \
+				.range = DT_INST_ENUM_IDX(n, nordic_range),                                   \
+			},                                                                         	      \
+		.frequency = DT_INST_PROP(n, nordic_frequency),                                       \
+		.out_div = DT_INST_PROP(n, nordic_out_div),                                           \
```

Comment:
Add a build assert for this one, it must be more than 0 from what I can see in the get_rate API implementation :)